src/shared.js

```diff
@@ -1583,6 +1585,38 @@ export const updateCollectionSettingData = async (biospecimenData, tubes, partic
             }
         });
     }
+    else if (settings[visit][conceptIds.baseline.bloodCollectedTime] !== '' ||  settings[visit][conceptIds.clinicalDashboard.bloodCollectedTime] !== ''){
+        const participantBloodCollected = participantData[conceptIds.baseline.bloodCollected] === conceptIds.yes;
+        const collectionSetting = biospecimenData[conceptIds.collection.collectionSetting];
+        const isResearch = collectionSetting === conceptIds.research;
+        const isClinical = collectionSetting === conceptIds.clinical;
+        let totalBloodTubesAvail = 6; // 6 blood tubes are available. Update this number if more blood tubes are added
+        bloodTubes.forEach(tube => {
+            const tubeIsNotCollected = biospecimenData[tube.concept][conceptIds.collection.tube.isCollected] === conceptIds.no;
+            if (tubeIsNotCollected && participantBloodCollected) {
+                totalBloodTubesAvail -= 1 // counter keeps track of unchecked blood tubes
+                if (totalBloodTubesAvail === 0) { // derived variables & timestamp are updated only if all the blood tubes are unchecked
+                    delete settings[visit][conceptIds.bloodCollectionSetting];
+                    if (isResearch) {
+                        delete settings[visit][conceptIds.baseline.bloodCollectedTime];
+                    }
+                    else if (isClinical) {
+                        settings[visit][conceptIds.clinicalDashboard.bloodCollected] = conceptIds.no;
+                        delete settings[visit][conceptIds.clinicalDashboard.bloodCollectedTime];
+
+                        if (urineTubesLength === 0 && mouthwashTubesLength === 0) { // the anySpecimenCollected variable will only be updated to NO if mouthwash & urine specimens are not present.
+                            settings[visit][conceptIds.anySpecimenCollected] = conceptIds.no;
+                            if (!(settings[visit][conceptIds.anySpecimenCollectedTime])) {
+                                delete settings[visit][conceptIds.anySpecimenCollectedTime];
+                            }
+                        }
+                    }
+                    derivedVariables[conceptIds.baseline.bloodCollected] = conceptIds.no;
+                    bloodTubesLength = totalBloodTubesAvail;
+                }
+            }
+        })
+    }
```

Comment:
This comment is referring to line 1713, I'm not sure we need to check the length of Object.keys() being greater than 0. If it's empty I don't think it will negatively impact things.